File: ChatGPT/Python/FactoryMethod/round3/source/coffee_factory.py
from abc import ABC, abstractmethod

# Abstract Product
class Coffee(ABC):
    @abstractmethod
    def get_description(self):
        pass

# Concrete Products
class Espresso(Coffee):
    def get_description(self):
        return "Espresso"

class Cappuccino(Coffee):
    def get_description(self):
        return "Cappuccino"

class Latte(Coffee):
    def get_description(self):
        return "Latte"

class Mocha(Coffee):
    def get_description(self):
        return "Mocha"

class Americano(Coffee):
    def get_description(self):
        return "Americano"

# Abstract Factory
class CoffeeFactory(ABC):
    @abstractmethod
    def create_coffee(self):
        pass

# Concrete Factories
class EspressoFactory(CoffeeFactory):
    def create_coffee(self):
        return Espresso()

class CappuccinoFactory(CoffeeFactory):
    def create_coffee(self):
        return Cappuccino()

class LatteFactory(CoffeeFactory):
    def create_coffee(self):
        return Latte()

class MochaFactory(CoffeeFactory):
    def create_coffee(self):
        return Mocha()

class AmericanoFactory(CoffeeFactory):
    def create_coffee(self):
        return Americano()

File: src/main/java/ChatGPT/Factory_Method/Round3/CoffeeFactory.java
package ChatGPT.Factory_Method.Round3;

// CoffeeFactory.java
class CoffeeFactory {
    public Coffee createCoffee(String type) {
        switch (type.toLowerCase()) {
            case "americano":
                return new Americano();
            case "latte":
                return new Latte();
            case "cappuccino":
                return new Cappuccino();
            case "espresso":
                return new Espresso();
            case "mocha":
                return new Mocha();
            default:
                throw new IllegalArgumentException("Unknown coffee type: " + type);
        }
    }
}

